### PR TITLE
Improve documentation related to Array error handling

### LINKED ISSUE #3834: Array warnings and errors are inconsistent, fail case for remove is not documented
**Your Godot version:**
3.2.2.rc1

**Issue description:**

When I run the following GDscript:

```GDscript
	var a = [10,11]
	print("a1: ",  a)
	a.remove(0)
	print("a2: ",  a)
	a.remove(10)
	print("a3: ", a)
	a.remove(0)
	print("a4: ", a)
	print("a5: ", a.front())
	print("a6: ", a.back())
	print("a7: ", a.pop_back())
	print("a8: ", a[5])
	print("Does not get here")
```

The output is:

> a1: [10, 11]
> a2: [11]
>  \* **./core/cowdata.h:161 - Index p_index = 10 is out of bounds (size() = 1).**
> a3: [11]
> a4: []
>  \* **Can't take value from empty array.**
> a5: Null
>  \* **Can't take value from empty array.**
> a6: Null
> a7: Null
>  \* **res://immediate_code.gd:30 - Invalid get index '5' (on base: 'Array').**

The documentation for the functions `front()`, `back()`, `pop_front()` and `pop_back()` all say: `Returns null if the array is empty.` However, a red warning message is printed for `front()` but not `pop_front()`.  Why the warning if this is documented behavior?

The biggest issue is `Array.remove()` with an index outside the range, which sometimes (not always), prints out a cryptic message from `core/cowdata.h`.

The documentation does not say what will happen (warning? crash?) if the index in `remove()` is outside of the range.  

Finally, when you access an element by an index outside the range, like `a[0]` on an empty array, you get an immediate crash.  This means that `a.front()` and `a[0]` do different things.  

I've looked into the code and there are uses of the macros `ERR_FAIL_INDEX` that prints a warning and `CRASH_BAD_INDEX` that generates a trap in some low-level code which is used in a lot of other places, not just the GDscript Array class.

It is not clear that the fail behaviors for these six Array functions are the most obvious, but what they do should be documented precisely.

**URL to the documentation page:**
https://docs.godotengine.org/en/stable/classes/class_array.html

### PR DESCRIPTION
**Pull request by Calinou**
This closes https://github.com/godotengine/godot-docs/issues/3834.

### CODE REVIEW

File: doc/classes/Array.xml [hunk 1]
```diff
@@ -283,14 +285,14 @@
...
-				Removes and returns the last element of the array. Returns [code]null[/code] if the array is empty.
+				Removes and returns the last element of the array. Returns [code]null[/code] if the array is empty (without printing an error message).
 			</description>
 		</method>
 		<method name="pop_front">
 			<return type="Variant">
 			</return>
 			<description>
-				Removes and returns the first element of the array. Returns [code]null[/code] if the array is empty.
+				Removes and returns the first element of the array. Returns [code]null[/code] if the array is empty (without printing an error message).
```
Comment (NathanLovato) [inline doc/classes/Array.xml:295]:
```suggestion
				Removes and returns the first element of the array. Returns [code]null[/code] if the array is empty, without printing an error message.
```

File: doc/classes/Array.xml [hunk 2]
```diff
@@ -317,7 +319,7 @@
 			<argument index="0" name="position" type="int">
 			</argument>
 			<description>
-				Removes an element from the array by index.
+				Removes an element from the array by index. Does nothing if the index does not exist in the given array.
```
Comment (NathanLovato) [inline doc/classes/Array.xml:322]:
```suggestion
				Removes an element from the array by index. If the index does not exist in the array, nothing happens.
```

File: doc/classes/Array.xml [hunk 3]
```diff
@@ -283,14 +285,14 @@
 			<return type="Variant">
 			</return>
 			<description>
-				Removes and returns the last element of the array. Returns [code]null[/code] if the array is empty.
+				Removes and returns the last element of the array. Returns [code]null[/code] if the array is empty (without printing an error message).
```
Comment (NathanLovato) [inline doc/classes/Array.xml:288]:
```suggestion
				Removes and returns the last element of the array. Returns [code]null[/code] if the array is empty, without printing an error message.
```

File: doc/classes/Array.xml [hunk 4]
```diff
@@ -119,7 +119,8 @@
 			<return type="Variant">
 			</return>
 			<description>
-				Returns the last element of the array. Throws an error and returns [code]null[/code] if the array is empty.
+				Returns the last element of the array. Prints an error and returns [code]null[/code] if the array is empty.
+				[b]Note:[/b] This is not the same as [code]array[-1][/code], as accessing by index will pause project execution when running from the editor if the array is empty.
```
Comment (NathanLovato) [inline doc/classes/Array.xml:123]:
```suggestion
				[b]Note:[/b] Calling this function is not the same as writing [code]array[-1][/code]. If the array is empty, accessing by index will pause project execution when running from the editor.
```

File: doc/classes/Array.xml [hunk 5]
```diff
@@ -216,7 +217,8 @@
 			<return type="Variant">
 			</return>
 			<description>
-				Returns the first element of the array. Throws an error and returns [code]null[/code] if the array is empty.
+				Returns the first element of the array. Prints an error and returns [code]null[/code] if the array is empty.
+				[b]Note:[/b] This is not the same as [code]array[0][/code], as accessing by index will pause project execution when running from the editor if the array is empty.
```
Comment (NathanLovato) [inline doc/classes/Array.xml:221]:
```suggestion
				[b]Note:[/b] Calling this function is not the same as writing [code]array[0][/code]. If the array is empty, accessing by index will pause project execution when running from the editor.
```

**Review by NathanLovato — CHANGES_REQUESTED** [review]
Some small changes to the docs to hopefully improve clarity a bit. I mostly made suggestions to remove the parentheses, following the docs writing guidelines, but I feel a change in wording can make some sentences a bit more explicit.

**Review by NathanLovato — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.3.

**Pull request merged**